.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "include/boost/compute"]
+	path = compute
+	url = https://github.com/boostorg/compute
```

Comment:
@guolinke Can you check for the licensing?

Boost licensing shouldn't be an issue as Boost users are free to do whatever they do with it. The only obligation is to not remove the Boost license from the submodule (if you create compiled code afterwards, then putting the Boost license is not required - it must be only there before compiling code).

Users are able to fetch the LightGBM repository without Boost using `recursive`, while those who want the full repository can use `recursive` on git.